pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGeneratorFunctors.hpp

```diff
@@ -371,86 +355,249 @@ void applyGenMultiRZ(
         });
 }
 
+template <class PrecisionT, class FuncT> class applyNCGenerator1Functor {
+    using KokkosIntVector = Kokkos::View<std::size_t *>;
+
+    Kokkos::View<Kokkos::complex<PrecisionT> *> arr;
+    const FuncT core_function;
+    KokkosIntVector indices;
+    KokkosIntVector parity;
+    KokkosIntVector rev_wires;
+    KokkosIntVector rev_wire_shifts;
+    std::size_t mask{0U};
+    static constexpr std::size_t one{1U};
+    std::size_t i0;
+    std::size_t i1;
+
+  public:
+    template <class ExecutionSpace>
+    applyNCGenerator1Functor([[maybe_unused]] ExecutionSpace exec,
+                             Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
+                             std::size_t num_qubits,
+                             const std::vector<std::size_t> &controlled_wires,
+                             const std::vector<bool> &controlled_values,
+                             const std::vector<std::size_t> &wires,
+                             FuncT core_function_)
+        : arr(arr_), core_function(core_function_) {
+        const std::size_t n_contr = controlled_wires.size();
+        const std::size_t n_wires = wires.size();
+        const std::size_t nw_tot = n_contr + n_wires;
+        PL_ASSERT(n_wires == 1);
+        PL_ASSERT(num_qubits >= nw_tot);
+
+        std::vector<std::size_t> all_wires;
+        all_wires.reserve(nw_tot);
+        all_wires.insert(all_wires.begin(), wires.begin(), wires.end());
```

Comment:
Could you please explain why both `wires` and `all_wires` are installed from the `all_wires.begin()`?

Reply:
I have swapped these around so we don't have to insert both from `all_wires.begin()`, which is bad for performance. I actually got this from [LQ ](https://github.com/PennyLaneAI/pennylane-lightning/blob/9fc9633b10fb653f63b770f51785b90e56d78fda/pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp#L1950), should I raise a ticket to swap it there too?